docs/source/notes/autograd.rst

```diff
@@ -210,3 +210,102 @@ No thread safety on C++ hooks
 Autograd relies on the user to write thread safe C++ hooks. If you want the hook
 to be correctly applied in multithreading environment, you will need to write
 proper thread locking code to ensure the hooks are thread safe.
+
+Autograd for Complex Numbers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What notion of complex derivative does PyTorch use?
+***************************************************
+
+PyTorch follows `JAX's <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`_
+convention for autograd for Complex Numbers.
+
+Suppose we have a function :math:`F: C -> C` which we can decompose into functions u and v
+which compute the real and imaginary parts of the function:
+
+    .. code::
+
+        def F(z):
+            x, y = real(z), imag(z)
+            return u(x, y) + v(x, y) * 1j
+
+The Jacobian, JVP and VJP for function :math:`F` at :math:`(x, y)` are defined as:
+
+    .. code::
+
+        def JVP(tangent):
+            c, d = real(tangent), imag(tangent)
+            return [1, i]^T * J * [c, d]
+
+    .. code::
+
+        def VJP(cotangent):
+            c, d = real(cotangent), imag(cotangent)
+            return [c, -d]^T * J * [1, -i]
+
+    where
+
+    .. math::
+
+        J = \begin{bmatrix}
+            \partial_0u(x, y) & \partial_1u(x, y)\\
+            \partial_0v(x, y) & \partial_1v(x, y) \end{bmatrix} \\
+
+In PyTorch, the VJP is mostly what we care about, as it is the computation performed when we do backward
+mode automatic differentiation. Notice that d and i (imaginary number) are negated in the formula above.
+
+Why is there a negative sign in the formula above?
+**************************************************
+
+For a function F: V → W, where are V and W are vector spaces. The output of
+the Vector-Jacobian Product :math:`VJP : V → (W^* → V^*)` is a linear map
```

Comment:
nit the VJP you define above is not this function here.
The one you define above is directly the linear mapping (because you hard coded the input in space V directly into J).

Reply:
oh yeah good catch updated!

Reply:
How was this updated?

Reply:
the "output of", I imagine